### DISCUSSION TITLE
How to keep history state of browser untouched? (to disable back gesture on iOS)

### DISCUSSION THREAD

**Discussion by 205g0** [sveltejs/kit#8774] [Q&A] [answered]
Since the back gesture on iOS Safari destroys every SPA UI, I'd like to keep history always the same despite having `goto()` calls. What's the most elegant way to achieve this with SvelteKit?  

Edit and fwiw: ofc iOS back gesture isn't disabled with this move but it wouldn't lead the last page but to the prior site outside of my SPA which I'd be ok with

**Comment by teemingc** [comment, marked as answer]
Something like this I think. Never tried it myself but it seems right.
```js
goto("/another-page", { replaceState: true })
```
> If `true`, will replace the current `history` entry rather than creating a new one with `pushState`

https://kit.svelte.dev/docs/modules#$app-navigation-goto

Similar to https://developer.mozilla.org/en-US/docs/Web/API/History/replaceState

**Reply by 205g0** [reply]
This did the trick, you're great, thanks!
